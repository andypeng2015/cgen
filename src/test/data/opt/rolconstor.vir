module foo

export function w $foo(w %x) {
@start:
  %w = lsl.w %x, 5_w
  %y = lsr.w %x, 27_w
  %z = or.w %w, %y
  ret %z
}
