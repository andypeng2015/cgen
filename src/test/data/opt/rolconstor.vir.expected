module foo

export function w $foo(w %x) {
@0:
  %0 = rol.w %x, 0x5_w ; @4
  ret %0
}
